S/Sundials/Sundials@7/build_tarballs.jl

```diff
@@ -0,0 +1,100 @@
+using BinaryBuilder
+
+name = "Sundials"
+version = v"7.4.0"
+
+# Collection of sources required to build Sundials
+sources = [
+    ArchiveSource("https://github.com/LLNL/sundials/releases/download/v$(version)/sundials-$(version).tar.gz",
+                  "679ddacdd77610110e613164e8297d6d0cd35bae8e9c3afc8e8ff6f99a1c2a7b"),
+    DirectorySource("./bundled"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/sundials*
+
+# Note: The SundialsSetupCompilers patch for 64-bit integer support
+# is no longer needed in v7.4.0 as the Fortran name mangling has been
+# restructured in the newer version of SUNDIALS.
+
+# Set up CFLAGS
+cd $WORKSPACE/srcdir/sundials*/cmake/tpl
+if [[ "${target}" == *-mingw* ]]; then
+    BLAS_NAME="${libdir}/libblastrampoline-5.${dlext}"
+    LAPACK_NAME="${libdir}/libblastrampoline-5.${dlext}"
+    atomic_patch -p2 $WORKSPACE/srcdir/patches/Sundials_windows.patch
+    # Work around https://github.com/LLNL/sundials/issues/29
+    # When looking for KLU libraries, CMake searches only for import libraries,
+    # this patch ensures we look also for shared libraries.
+    atomic_patch -p2 $WORKSPACE/srcdir/patches/Sundials_findklu_suffixes.patch
+else
+    BLAS_NAME="${libdir}/libblastrampoline.${dlext}"
+    LAPACK_NAME="${libdir}/libblastrampoline.${dlext}"
+fi
+
+# Build
+cd $WORKSPACE/srcdir/sundials*
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}" \
+    -DEXAMPLES_ENABLE_C=OFF \
+    -DENABLE_KLU=ON \
+    -DKLU_INCLUDE_DIR="${includedir}/suitesparse" \
+    -DKLU_LIBRARY_DIR="${libdir}" \
+    -DKLU_WORKS=ON \
+    -DENABLE_LAPACK=ON \
+    -DBLAS_LIBRARIES="${BLAS_NAME}" \
+    -DLAPACK_LIBRARIES="${LAPACK_NAME}" \
```

Comment:
Setting these variables directly is actually not supported by CMake (https://gitlab.kitware.com/cmake/cmake/-/issues/25913#note_1511382).

Instead, the build script should remove the image CMake version (`apk del cmake`) and then add a `HostBuildDependency("CMake_jll")` and use `-DBLA_VENDOR="libblastrampoline"` in the CMake command instead.

Reply:
We follow what the Sundials docs say: https://sundials.readthedocs.io/en/latest/sundials/Install_link.html

Reply:
The Sundials docs are fairly generic, and they actually seem to suggest just leaving all those variables blank and letting CMake auto-detect the library (they don't preset any of those variables). Their docs should really be suggesting that people use the `BLA_VENDOR` CMake flag, because upstream CMake has said that overwriting those variables is actually a bad idea (we did see problems doing this in a Ygg recipe a while ago, hence why I added LBT to CMake upstream).